src/parser/mod.rs

```diff
@@ -10133,6 +10134,30 @@ impl<'a> Parser<'a> {
                 columns,
                 alias,
             })
+        } else if self.parse_keyword_with_tokens(Keyword::OPENJSON, &[Token::LParen]) {
+            let json_expr = self.parse_expr()?;
+            let json_path = if self.consume_token(&Token::Comma) {
+                Some(self.parse_value()?)
+            } else {
+                None
+            };
+            self.expect_token(&Token::RParen)?;
+            let columns = if self.parse_keyword(Keyword::WITH) {
+                self.expect_token(&Token::LParen)?;
+                let columns =
+                    self.parse_comma_separated(Parser::parse_openjson_table_column_def)?;
+                self.expect_token(&Token::RParen)?;
+                columns
+            } else {
+                Vec::new()
+            };
+            let alias = self.parse_optional_table_alias(keywords::RESERVED_FOR_TABLE_ALIAS)?;
+            Ok(TableFactor::OpenJsonTable {
+                json_expr,
+                json_path,
+                columns,
+                alias,
+            })
```

Comment:
One request (I think the other branches in this function don't do a good job of it :) could we move this to its own function e.g `self.parse_open_json_table_factor()` and call that function from this branch instead? thinking in order to keep the branches shorter/more-focused